### PR TITLE
Feat: Adds `LoadingButton` and `Loader` component

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
- [x] Creates `Loader` component on @faststore/components 
- [x] Adds `Loader` styles on @faststore/ui
- [x] Updates `Button` to receive a Loading state
- [x] Adds `Loader` styles on @faststore/ui
- [x] Removes `Spinner` component
- [x] Removes `LoadingButton` component

## How to test it?

run yarn start in app/docs folder and checks the `Loader` and `Button` components or test it using the doc preview link:
- [Loader Component](https://faststore-site-git-feat-loading-button-fs-751-faststore.vercel.app/components/atoms/loader)
- [Button Component](https://faststore-site-git-feat-loading-button-fs-751-faststore.vercel.app/components/atoms/button)

### CODE REVIEW

File: apps/site/pages/components/atoms/button.mdx [hunk 1]
```diff
@@ -603,6 +604,56 @@ import '@faststore/ui/src/components/atoms/Button/styles.scss'
...
+  <Tab>
+
+    ```tsx
+    const [loading, setLoading] = useState(false)
+
+    <Button
+      variant="primary"
+      onClick={() => {
+        setLoading(true)
+      }}
```
Comment (emersonlaurentino) [inline apps/site/pages/components/atoms/button.mdx:639]:
```suggestion
      onClick={() => setLoading(true)}
```

File: packages/components/src/atoms/Loader/Loader.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const Loader = forwardRef<HTMLDivElement, LoaderProps>(function Loader(
+  {
+    variant = 'dark',
+    testId = 'fs-loader',
+    ...otherProps
+  }: LoaderProps,
+  ref
+) {
+  return (
+    <div ref={ref} data-fs-loader data-fs-loader-variant={variant} data-testid={testId} {...otherProps}>
```
Comment (emersonlaurentino) [inline packages/components/src/atoms/Loader/Loader.tsx:23]: I think you need to run prettier in this file

File: packages/components/src/molecules/BuyButton/BuyButton.tsx [hunk 3]
```diff
@@ -13,8 +13,7 @@ function BuyButton({
   ...otherProps
 }: BuyButtonProps) {
   return (
-    <Button data-fs-buy-button data-testid={testId} {...otherProps}>
-      <Icon component={<ShoppingCart />} />
+    <Button data-fs-buy-button icon={<ShoppingCart />} iconPosition="left" data-testid={testId} {...otherProps}>
```
Comment (emersonlaurentino) [inline packages/components/src/molecules/BuyButton/BuyButton.tsx:16]: in this one too

File: apps/site/components/Button/ButtonLoading.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+const ButtonLoading = () => {
+  const [loading, setLoading] = useState(false)
+
+  return (
+    <Button variant="primary" onClick={() => { setLoading(true) }} loadingLabel="Loading" loading={loading}>
+      Click me to trigger loading
+    </Button>
+  )
+
+}
```
Comment (emersonlaurentino) [inline apps/site/components/Button/ButtonLoading.tsx:13]: What do you think?


https://user-images.githubusercontent.com/10627086/212949574-4fd73379-2473-4352-bd64-a413a45524af.mov




```suggestion
import React, { useEffect, useState } from 'react'
import { Button } from '@faststore/ui'

const ButtonLoading = () => {
  const [loading, setLoading] = useState(false)

  useEffect(() => {
    let timer = null

    if (loading) {
      timer = window.setInterval(() => setLoading(false), 2000)
    }

    return () => {
      window.clearInterval(timer)
    }
  }, [loading])

  return (
    <Button
      variant="primary"
      onClick={() => setLoading(true)}
      loadingLabel="Loading"
      loading={loading}
    >
      Click me to trigger loading
    </Button>
  )
}
```

File: apps/site/pages/components/atoms/button.mdx [hunk 5]
```diff
@@ -603,6 +604,54 @@ import '@faststore/ui/src/components/atoms/Button/styles.scss'
...
 
+#### Loading
+
+<Tabs items={['Example', 'Code']} defaultIndex="0">
+  <Tab>
+    <OverviewSection direction="column">
+      <div data-doc-overview-group>
+        <h5 data-doc-overview-group-title>No label</h5>
+        <Button variant="primary" loading>
+          Buy
```
Comment (emersonlaurentino) [inline apps/site/pages/components/atoms/button.mdx:615]:
```suggestion
         {'Buy'}
```

**Review by emersonlaurentino — APPROVED** [review]

File: apps/site/pages/components/atoms/loader.mdx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  <Loader variant="light" />
+</OverviewSection>
+```tsx
+<Loader variant="light" />
+```
+
+<TokenTable>
+  <TokenRow
+    token="--fs-loader-light-item-bkg-color"
+    value="var(--fs-color-neutral-0)"
```
Comment (hellofanny) [inline apps/site/pages/components/atoms/loader.mdx:146]: do u think we can use semantical token here?
`--fs-color-tertiary-bkg-light`

File: apps/site/pages/components/molecules/rating.mdx [hunk 7]
```diff
@@ -135,3 +135,21 @@ const [rating, setRating] = useState(3)
 <TokenTable>
   <TokenRow token="--fs-rating-actionable-gap" value="0" />
 </TokenTable>
+
+---
+
+## Customization
```
Comment (hellofanny) [inline apps/site/pages/components/molecules/rating.mdx:141]: nice heheh :)

**Review by hellofanny — APPROVED** [review]
LGTM!! 🌟

File: apps/site/components/Button/ButtonLoading.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import { Button } from '@faststore/ui'
+
+const ButtonLoading = () => {
+  const [loading, setLoading] = useState(false)
+
+  useEffect(() => {
+    let timer = null
+
+    if (loading) {
+      timer = window.setInterval(() => setLoading(false), 2000)
```
Comment (eduardoformiga) [inline apps/site/components/Button/ButtonLoading.tsx:11]: I think we could use `setTimeout` instead.

File: apps/site/components/Button/ButtonLoading.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    return () => {
+      window.clearInterval(timer)
```
Comment (eduardoformiga) [inline apps/site/components/Button/ButtonLoading.tsx:15]: and clearTimeout

File: packages/components/src/atoms/Loader/index.ts [hunk 10]
```diff
@@ -0,0 +1,2 @@
+export { default } from './Loader'
+export type { LoaderProps } from './Loader'
```
Comment (eduardoformiga) [inline packages/components/src/atoms/Loader/index.ts:2]: Screen Shot 2023-01-18 at 20 08 01

GH is complaining.

```suggestion
export type { LoaderProps } from './Loader'

```

File: packages/ui/src/components/atoms/Loader/styles.scss [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    opacity: var(--fs-loader-item-initial-opacity);
+    top: 0;
+    position: relative;
+    animation: load 1.2s infinite var(--fs-loader-animation-function) calc(var(--fs-loader-item-delay,0)*var(--fs-loader-animation-timing));
+
+    &:first-of-type { --fs-loader-item-delay: -1; }
+    &:last-of-type  { --fs-loader-item-delay:  1; }
+  }
+
+  @keyframes load {
```
Comment (eduardoformiga) [inline packages/ui/src/components/atoms/Loader/styles.scss:44]: 💯

File: packages/ui/src/styles/components.scss [hunk 12]
```diff
@@ -3,8 +3,9 @@
 @import "../components/atoms/Button/styles";
 @import "../components/atoms/Checkbox/styles";
 @import "../components/atoms/Icon/styles";
-@import "../components/atoms/Link/styles";
 @import "../components/atoms/Input/styles";
+@import "../components/atoms/Loader/styles";
```
Comment (eduardoformiga) [inline packages/ui/src/styles/components.scss:7]: ⛏️   could you please sort it alphabetically?

**Review by eduardoformiga — APPROVED** [review]
Amazing Job! 🚀  Just left a few comments.

**Pull request merged**